[Inductor][Triton] Fix SCALING_ROWWISE misclassification for scalar scales

NikhilAPatel opened the pull request:
Summary: In `tuned_scaled_mm()`, we unsqeeuze any scalar scale from [] -> [1, 1]. Later, when we are determining how to set the `SCALING_ROWWISE` kernel attribute, we check whether the scale has 2 dimensions. However, since we previously unsqueezed any scalar scales, this will always evaluate to True.

Test Plan:
Run the following tests in test/inductor/test_fp8.py:
test_tensorwise_scaling_tma_template
test_rowwise_scaling_tma_template

Rollback Plan:

Differential Revision: D80108117




cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben

ngimel commented on test/inductor/test_fp8.py:
```diff
@@ -450,13 +452,18 @@ def linear(x_fp8, x_inverse_scale, w_t_fp8, w_inverse_scale, bias):
             )
+
+            src = code[0]
+            if "triton_tem_fused__scaled_mm" in src:
```
this is a bit fragile, if naming convention changes this test won't check anything

eellison reviewed (approved):
thank you!

eellison commented:
ready to land ?

NikhilAPatel commented:
@eellison Almost, there's a failing test: https://github.com/pytorch/pytorch/actions/runs/17111066429/job/48533904007?pr=160450. I'm not able to repro the error locally. Do you have any insight to what may be causing it to fail?

NikhilAPatel committed:
[Inductor][Triton] Fix SCALING_ROWWISE misclassification for scalar scales (#160450)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/160450

In `tuned_scaled_mm()`, we unsqeeuze any scalar scale from [] -> [1, 1]. Later, when we are determining how to set the `SCALING_ROWWISE` kernel attribute, we check whether the scale has 2 dimensions. However, since we previously unsqueezed any scalar scales, this will always evaluate to True.

Test Plan:
`
buck2 test 'fbcode//mode/opt' fbcode//caffe2/test/inductor:fp8 -- test_tensorwise_scaling
`

`
buck2 test 'fbcode//mode/opt' fbcode//caffe2/test/inductor:fp8 -- test_rowwise_scaling
`

`
buck2 test 'fbcode//mode/opt' fbcode//caffe2/test/inductor:fp8 -- test_tensorwise_scaling_acceptable_input_dims
`

`
buck2 test 'fbcode//mode/opt' fbcode//caffe2/test/inductor:fp8 -- test_tensorwise_scaling_acceptable_input_dims
`

Rollback Plan:

Reviewed By: eellison, PaulZhang12

Differential Revision: D80108117
